tests/dataimportcron_test.go

```diff
@@ -28,90 +29,118 @@ const (
+			By(fmt.Sprintf("Wait for DataImportCron Progressing:%s, UpToDate:%s", statusProgressing, statusUpToDate))
 			Eventually(func() bool {
-				_, err = f.K8sClient.BatchV1beta1().CronJobs(f.CdiInstallNs).Get(context.TODO(), controller.GetCronJobName(cron), metav1.GetOptions{})
-				if errors.IsNotFound(err) {
-					return false
-				}
-				return err == nil
+				var err error
+				cron, err = f.CdiClient.CdiV1beta1().DataImportCrons(ns).Get(context.TODO(), cronName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				condProgressing := controller.FindDataImportCronConditionByType(cron, cdiv1.DataImportCronProgressing)
+				condUpToDate := controller.FindDataImportCronConditionByType(cron, cdiv1.DataImportCronUpToDate)
+				return condProgressing != nil && condProgressing.Status == statusProgressing &&
+					condUpToDate != nil && condUpToDate.Status == statusUpToDate
 			}, dataImportCronTimeout, pollingInterval).Should(BeTrue())
 		}
 
 		var lastImportDv, currentImportDv string
 		for i := 0; i < repeat; i++ {
+			By(fmt.Sprintf("Iter #%d", i))
 			if i > 0 {
-				// Emulate source update using digests from https://quay.io/repository/kubevirt/cirros-container-disk-demo?tab=tags
-				digest := []string{
-					"sha256:68b44fc891f3fae6703d4b74bcc9b5f24df8d23f12e642805d1420cbe7a4be70",
-					"sha256:90e064fca2f47eabce210d218a45ba48cc7105b027d3f39761f242506cad15d6",
-				}[i%2]
-
-				By(fmt.Sprintf("Update source desired digest to %s", digest))
-				Eventually(func() bool {
+				if createErrorDv {
+					By("Set desired digest to nonexisting one")
 					cron, err = f.CdiClient.CdiV1beta1().DataImportCrons(ns).Get(context.TODO(), cronName, metav1.GetOptions{})
 					Expect(err).ToNot(HaveOccurred())
-					if cron.Annotations == nil {
-						cron.Annotations = make(map[string]string)
-					}
-					cron.Annotations[controller.AnnSourceDesiredDigest] = digest
+					cron.Annotations[controller.AnnSourceDesiredDigest] = "sha256:12345678900987654321"
 					cron, err = f.CdiClient.CdiV1beta1().DataImportCrons(ns).Update(context.TODO(), cron, metav1.UpdateOptions{})
-					return err == nil
-				}, dataImportCronTimeout, pollingInterval).Should(BeTrue())
-			}
-			By("Wait for CurrentImports DataVolumeName update")
-			Eventually(func() bool {
-				cron, err = f.CdiClient.CdiV1beta1().DataImportCrons(ns).Get(context.TODO(), cronName, metav1.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				if len(cron.Status.CurrentImports) == 0 {
-					return false
+					Expect(err).ToNot(HaveOccurred())
+
+					By("Wait for CurrentImports update")
+					Eventually(func() bool {
+						cron, err = f.CdiClient.CdiV1beta1().DataImportCrons(ns).Get(context.TODO(), cronName, metav1.GetOptions{})
+						currentImportDv = cron.Status.CurrentImports[0].DataVolumeName
+						Expect(currentImportDv).ToNot(BeEmpty())
+						return currentImportDv != lastImportDv
+					}, dataImportCronTimeout, pollingInterval).Should(BeTrue())
```

Comment:
why don't you return currentImportDv and directly compare in Should
`ShouldNot(BeEqual(lastImportDv))`

Reply:
Nice, but isn't `currentImportDv != lastImportDv` more readable in this case?

Reply:
well when I read text quickly I tend to check all Assertions/expectatinos/Should parts,
and here I can see `Eventually` `return bool` `BeTrue`, after changing, I will see `Eventually` `return currentImportDv` `Be Equal` `lastImportDv`. What is more, the message on failure will be better. It will show both Names.

Message will say: `Eexpected name1 to be equal to name2`, instead of `Expected <bool>: false to be true`

Reply:
Right, fixing.